src/main/java/org/gridsuite/ds/server/service/parameters/impl/ParametersServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+import com.powsybl.dynawaltz.DynaWaltzParameters;
+import org.gridsuite.ds.server.service.parameters.ParametersService;
+import org.gridsuite.ds.server.utils.xml.XmlMerge;
+import org.gridsuite.ds.server.utils.xml.implementation.SimpleXmlMerge;
+import org.springframework.stereotype.Service;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+/**
+ * @author Thang PHAM <quyet-thang.pham at rte-france.com>
+ */
+@Service
+public class ParametersServiceImpl implements ParametersService {
+
+    @Override
+    public byte[] getEventModel() throws IOException {
+        // read the events.groovy in the "parameters" resources
+        return getClass().getResourceAsStream(Paths.get(PARAMETERS_DIR, EVENTS_GROOVY).toString()).readAllBytes();
+    }
+
+    @Override
+    public byte[] getCurveModel() throws IOException {
+        // read the curves.groovy in the "parameters" resources
+        return getClass().getResourceAsStream(Paths.get(PARAMETERS_DIR, CURVES_GROOVY).toString()).readAllBytes();
+    }
+
+    @Override
+    public DynamicSimulationParameters getDynamicSimulationParameters(byte[] dynamicParams) throws IOException, RuntimeException {
+
+        // prepare a temp dir for current running simulation
+        Path configDir = PlatformConfig.defaultConfig().getConfigDir().orElseThrow();
+        Path workingDir = Files.createTempDirectory(configDir, WORKING_DIR_PREFIX);
+
+        // merge dynamicParams with events.par then load parametersFile in a runtime tmp directory
+        InputStream eventsIs = getClass().getResourceAsStream(Paths.get(PARAMETERS_DIR, EVENTS_PAR).toString());
+        byte[] eventsParams = eventsIs.readAllBytes();
+        if (dynamicParams.length != 0 && eventsParams.length != 0) { // suppose two models .par are valid => do merge
+            mergeParams(dynamicParams, eventsParams, workingDir, MODELS_PAR);
+        } else { // create without events par
+            Files.copy(new ByteArrayInputStream(dynamicParams), workingDir.resolve(MODELS_PAR));
+        }
+
+        // load two others files
+        for (String parFileName : List.of(NETWORK_PAR, SOLVERS_PAR)) {
+            Files.copy(getClass().getResourceAsStream(Paths.get(PARAMETERS_DIR, parFileName).toString()), workingDir.resolve(parFileName));
+        }
+
+        // load parameter file then config paths
+        DynamicSimulationParameters parameters = JsonDynamicSimulationParameters.read(getClass().getResourceAsStream(Paths.get(PARAMETERS_DIR, PARAMETERS_JSON).toString()));
+        DynaWaltzParameters dynaWaltzParameters = parameters.getExtension(DynaWaltzParameters.class);
```

Comment:
To keep in mind to better design the dynamic simulation API and dynamo implementation on PowSyBl side: we should need to do that and create an explicit dependency to Dynawaltz class and keep dynamic simulation abstraction all over this micro service. But we are probably very far from it.

Reply:
Noted by comments directly in the code